### PR TITLE
Shugeo dropout bp

### PR DESCRIPTION
**Pull request by shugeo**
with dropout_bp op and tests.

### CODE REVIEW

File: libnd4j/include/ops/declarable/headers/parity_ops.h [hunk 1]
```diff
@@ -751,6 +751,9 @@ namespace nd4j {
         #if NOT_EXCLUDED(OP_dropout)
         DECLARE_CONFIGURABLE_OP(dropout, 1, 1, true, 1, 1);
         #endif
+        #if NOT_EXCLUDED(OP_dropout_bp)
```
Comment (raver119) [inline libnd4j/include/ops/declarable/headers/parity_ops.h]: All _bp ops must be tied to ff ops

File: libnd4j/tests_cpu/layers_tests/DeclarableOpsTests9.cpp [hunk 2]
```diff
@@ -479,7 +479,177 @@ TEST_F(DeclarableOpsTests9, tile_bp_test6) {
...
+
+    auto ressX = op2.execute({&x1, &x0}, {0.5f}, {119});
+
+    ASSERT_EQ(ND4J_STATUS_OK, ressX->status());
+//    ressX->at(0)->printIndexedBuffer("02Dropout result is ");
+//    ASSERT_TRUE(exp.equalsTo(ressX->at(0)));
```
Comment (raver119) [inline libnd4j/tests_cpu/layers_tests/DeclarableOpsTests9.cpp]: So, what are we testing here?

File: libnd4j/tests_cpu/layers_tests/DeclarableOpsTests9.cpp [hunk 3]
```diff
@@ -479,7 +479,177 @@ TEST_F(DeclarableOpsTests9, tile_bp_test6) {
...
+    Nd4jLong* _bufferA = new Nd4jLong[100000];
+    long _seed = 119L;
+    auto _rngA = (nd4j::random::RandomBuffer *) nativeOps.initRandom(nullptr, _seed, 100000, (Nd4jPointer) _bufferA);
+
+    float prob[] = {0.5f, 0.5f, 1.5f, 1.6f};
+    
+    x0.template applyRandom<randomOps::AlphaDropOut<float>>(_rngA, nullptr, &x0, prob);
+//    x1.template applyRandom<randomOps::DropOut<float>>(_rngB, nullptr, &x1, prob);
+//    x0.printIndexedBuffer("Result1Alpha");
+//    ASSERT_TRUE(x0.equalsTo(&x1));
```
Comment (raver119) [inline libnd4j/tests_cpu/layers_tests/DeclarableOpsTests9.cpp]: Same question. What this test tests?

File: libnd4j/tests_cpu/layers_tests/DeclarableOpsTests9.cpp [hunk 4]
```diff
@@ -479,7 +479,177 @@ TEST_F(DeclarableOpsTests9, tile_bp_test6) {
...
+//    NDArrayFactory<float>::linspace(1, x1);
+    NativeOps nativeOps;
+
+    Nd4jLong* _bufferA = new Nd4jLong[100000];
+    long _seed = 119L;
+    auto _rngA = (nd4j::random::RandomBuffer *) nativeOps.initRandom(nullptr, _seed, 100000, (Nd4jPointer) _bufferA);
+
+    float prob[] = {0.5f, 1.0f, 0.f, 0.f};
+    
+    x0.template applyRandom<randomOps::AlphaDropOut<float>>(_rngA, nullptr, &x0, prob);
```
Comment (raver119) [inline libnd4j/tests_cpu/layers_tests/DeclarableOpsTests9.cpp]: Same question as above.

**Comment by shugeo** [comment]
working with node random generator.

**Comment by agibsonccc** [comment]
Where are we here?

**Comment by raver119** [comment]
This PR won't be merged before release: it involves new RNG for Graph, which better to be extensively tested via snapshots etc.

**Pull request closed without merging**